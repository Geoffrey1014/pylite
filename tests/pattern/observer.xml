lambda
  name __init__
  param_count 0
  local_count 0
  label_count 0
  insn_vector
    new_class c0
    store_global c0
    load_const c1
    load_global c0
    store_field c2
    load_const c3
    load_global c0
    store_field c4
    load_const c5
    load_global c0
    store_field c6
    load_const c7
    load_global c0
    store_field c8
    new_class c9
    store_global c9
    load_global c0
    load_global c9
    inherit
    load_const c10
    load_global c9
    store_field c2
    load_const c11
    load_global c9
    store_field c12
    new_class c13
    store_global c13
    load_const c14
    load_global c13
    store_field c12
    new_class c15
    store_global c15
    load_const c16
    load_global c15
    store_field c12
    load_const c17
    store_global c18
    load_global c18
    call 0
    pop
    load_const c19
    return
  const_vector
    Subject
    lambda
      name __init__
      param_count 1
      local_count 0
      label_count 0
      insn_vector
        new_list 0
        load_local 0
        store_field c0
        load_const c1
        return
      const_vector
        observers
        None
      guard_vector
      line_number_map 0 3 3 2 2
    __init__
    lambda
      name attach
      param_count 2
      local_count 0
      label_count 0
      insn_vector
        load_local 1
        load_local 0
        load_field c0
        load_field c1
        call 1
        pop
        load_const c2
        return
      const_vector
        observers
        append
        None
      guard_vector
      line_number_map 6 6 6 6 6 6 5 5
    attach
    lambda
      name detach
      param_count 2
      local_count 0
      label_count 0
      insn_vector
        load_local 1
        load_local 0
        load_field c0
        load_field c1
        call 1
        pop
        load_const c2
        return
      const_vector
        observers
        remove
        None
      guard_vector
      line_number_map 9 9 9 9 9 9 8 8
    detach
    lambda
      name notify
      param_count 1
      local_count 2
      label_count 3
      insn_vector
        load_local 0
        load_field c0
        iterator
        store_local -2
        jmp L1
        label L0
        store_local -1
        load_local 0
        load_local -1
        load_field c1
        call 1
        pop
        label L1
        load_local -2
        next L0
        pop
        label L2
        load_const c2
        return
      const_vector
        observers
        update
        None
      guard_vector
      line_number_map 12 12 12 12 12 12 12 13 13 13 13 13 12 12 12 12 12 11 11
    notify
    Stock
    lambda
      name __init__
      param_count 2
      local_count 0
      label_count 0
      insn_vector
        load_local 0
        load_global c0
        load_field c1
        call 1
        pop
        load_local 1
        load_local 0
        store_field c2
        load_const c3
        return
      const_vector
        Subject
        __init__
        data
        None
      guard_vector
      line_number_map 17 17 17 17 17 18 18 18 16 16
    lambda
      name update
      param_count 2
      local_count 0
      label_count 0
      insn_vector
        load_local 1
        load_local 0
        store_field c0
        load_local 0
        load_field c1
        call 0
        pop
        load_const c2
        return
      const_vector
        data
        notify
        None
      guard_vector
      line_number_map 21 21 21 22 22 22 22 20 20
    update
    Tom
    lambda
      name update
      param_count 2
      local_count 0
      label_count 0
      insn_vector
        load_local 1
        load_field c0
        load_const c1
        mod
        print
        load_const c2
        return
      const_vector
        data
        "Tom: %d"
        None
      guard_vector
      line_number_map 26 26 26 26 26 25 25
    Jerry
    lambda
      name update
      param_count 2
      local_count 0
      label_count 0
      insn_vector
        load_local 1
        load_field c0
        load_const c1
        mod
        print
        load_const c2
        return
      const_vector
        data
        "Jerry: %d"
        None
      guard_vector
      line_number_map 30 30 30 30 30 29 29
    lambda
      name main
      param_count 0
      local_count 3
      label_count 0
      insn_vector
        load_const c0
        load_global c1
        call 1
        store_local -1
        load_global c2
        call 0
        store_local -2
        load_global c3
        call 0
        store_local -3
        load_local -2
        load_local -1
        load_field c4
        call 1
        pop
        load_local -3
        load_local -1
        load_field c4
        call 1
        pop
        load_const c5
        load_local -1
        load_field c6
        call 1
        pop
        load_const c7
        load_local -1
        load_field c6
        call 1
        pop
        load_const c8
        return
      const_vector
        100.000000
        Stock
        Tom
        Jerry
        attach
        110.000000
        update
        90.000000
        None
      guard_vector
      line_number_map 33 33 33 33 34 34 34 35 35 35 36 36 36 36 36 37 37 37 37 37 38 38 38 38 38 39 39 39 39 39 32 32
    main
    None
  guard_vector
  line_number_map 1 1 2 2 2 5 5 5 8 8 8 11 11 11 15 15 15 15 15 16 16 16 20 20 20 24 24 25 25 25 28 28 29 29 29 32 32 41 41 41 1 1
