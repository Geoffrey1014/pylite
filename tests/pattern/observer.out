Tom: 110
Jerry: 110
Tom: 90
Jerry: 90
